src/app/main.js

```diff
@@ -0,0 +1,74 @@
+import app from 'ampersand-app';
+import webChannel from './lib/web_channel';
+import ViewSwitcher from 'ampersand-view-switcher';
+
+import ExperimentsCollection from './collections/experiments';
+import Me from './models/me';
+import HomePage from './views/home_page';
+import Router from './lib/router';
+
+app.extend({
+  initialize() {
+    window.app = this;
+
+    // for now, instantiate the collection with dummy data
+    app.experiments = new ExperimentsCollection([
+      {
+        displayName: 'Universal Search',
+        name: 'universal-search',
+        isInstalled: false
+      }, {
+        displayName: 'Side Tabs',
+        name: 'side-tabs',
+        isInstalled: false
+      }, {
+        displayName: 'Snooze Tabs',
+        name: 'snooze-tabs',
+        isInstalled: false
+      }, {
+        displayName: 'Cheese Tabs',
+        name: 'cheese-tabs',
+        isInstalled: false
+      }
+    ]);
+    // empty (for now) user model
+    // trying out ampersand's 'me' convention, though I find it kinda dumb
+    app.me = new Me();
+
+    // ping the addon to see if it's installed
+    this.listenTo(webChannel, 'from-addon-to-web', (data) => alert(data));
+    webChannel.sendMessage('from-web-to-addon', { loaded: true });
+
+    // start with the home page view
+    const view = new HomePage({
+      experiments: app.experiments
+    });
+
+    // start router + history
+    app.router = new Router();
+    app.router.history.start();
+
+    app.pageContainer = document.querySelector('#container');
+    app.viewSwitcher = new ViewSwitcher(app.pageContainer, {
+      // do some nice things when you switch to a new page-level view
+      // inspired by ampersand-view-switcher readme
+      show: function showView(newView) {
+        // don't needlessly re-show a shown view
+        // works around ampersand-view-switcher bug #26 and seems like a reasonable requirement
+        if (newView === app.currentPage) {
```

Comment:
done: https://github.com/AmpersandJS/ampersand-view-switcher/issues/26#issuecomment-135269905

thanks for the encouragement :-)